src/drivers/imx/interrupt-generic.c

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright 2021 NXP
+//
+// Author: Peng Zhang <peng.zhang_8@nxp.com>
+
+#include <sof/common.h>
+#include <sof/drivers/interrupt.h>
+#include <sof/lib/cpu.h>
+#include <sof/lib/io.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/list.h>
+#include <sof/spinlock.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/* fa00558c-d653-4851-a03a-b21f125a9524 */
+DECLARE_SOF_UUID("generic-irq-imx", generic_irq_imx_uuid, 0xfa00558c, 0xd653, 0x4851,
+		 0xa0, 0x3a, 0xb2, 0x1f, 0x12, 0x5a, 0x95, 0x24);
+
+DECLARE_TR_CTX(noirq_i_tr, SOF_UUID(generic_irq_imx_uuid), LOG_LEVEL_INFO);
+
+int irqstr_get_sof_int(int irqstr_int)
```

Comment:
Since this is the generic interrupt, I believe this should be 
```
int platform_get_interrupt(int irq)
{
	return irq;
}
```
Don't mention irqstr (from irq steer) anymore. :)

Reply:
@TangleZ @iuliana-prodan I think Peng version is OK but we need some explanation

/* TODO: Fix this. Now this is required because ESAI driver assumes all i.MX8 platforms use irqsteer */